lib/util.go

```diff
@@ -0,0 +1,68 @@
+package docker2aci
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+type Compression int
+
+const (
+	Uncompressed Compression = iota
+	Gzip
+)
+
+func makeEndpointsList(headers []string) []string {
+	var endpoints []string
+
+	for _, ep := range headers {
+		endpointsList := strings.Split(ep, ",")
+		for _, endpointEl := range endpointsList {
+			endpoints = append(
+				endpoints,
+				// TODO(iaguis) discover if httpsOrHTTP
+				fmt.Sprintf("https://%s/v1/", strings.TrimSpace(endpointEl)))
+		}
+	}
+
+	return endpoints
+}
+
+func decompress(layer io.Reader) (io.Reader, error) {
+	bufR := bufio.NewReader(layer)
+	bs, _ := bufR.Peek(10)
+
+	compression := detectCompression(bs)
+	switch compression {
+	case Gzip:
+		gz, err := gzip.NewReader(bufR)
+		if err != nil {
+			return nil, fmt.Errorf("Error reading layer (gzip): %v", err)
+		}
+		return gz, nil
+	case Uncompressed:
+		return bufR, nil
+	default:
+		return nil, fmt.Errorf("Unknown layer format")
+	}
+}
```

Comment:
Hrm, these things are already in the aci libraries: http://godoc.org/github.com/appc/spec/aci#NewCompressedReader